starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +84,14 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    if (id != null & !id.equals("")) {
+      Query query = new Query();
+      query.addCriteria(Criteria.where("id").is(id));
+      Inventory inventory = this.mongoTemplate.findAndRemove(query, Inventory.class);
+
+      Optional<Inventory> deletedInv = Optional.ofNullable(inventory);
+      return deletedInv;
+    }
+    return null;
```

Comment:
Instead of the "if" statement, you can actually use Assert and hasText().
Also, you can use Optional to your advantage and just return that; you want to avoid returning null because it could cause issues later.
Otherwise, it looks good.